packages/react-interactions/events/src/dom/create-event-handle/useEvent.js

```diff
@@ -10,7 +10,7 @@
 import * as React from 'react';
 import * as ReactDOM from 'react-dom';
 
-const {useEffect, useRef} = React;
+const {useLayoutEffect, useRef} = React;
```

Comment:
Was the replacement of `useEffect` with `useLayoutEffect` made to keep timing consistent with what we used to do in `commitUnmount`? At first I thought maybe it was related to the recent passive effects timing changes.

Reply:
Actually, we have a `useEvent` on www and found that we need to mostly use `useLayoutEffect` to add events, otherwise you run into issues with `focus` and `blur` not getting captured because they happen before `useEffect` fires. I just made the change to the React repo for this too.

Reply:
Gotcha. That makes sense.

Reply:
> you run into issues with `focus` and `blur` not getting captured because they happen before useEffect fires

What are the specific issues?

Reply:
I assumed the issue was that e.g. a newly added element would be focusable before the "focus" event listener was actually attached (so the event would be missed)

Reply:
The were three cases that caused hundreds of internal tests to fail:
- when using the `autoFocus` prop on elements
- where we have user-land `focus()`, which is common in modals and rich text fields like DraftJS.
- there were some e2e test failures which seemed to be related to iframes

Switching to `useLayoutEffect` fixed all the above cases, and I don't imagine there to be any perf loss here either, so it seemed like a sensible change given the problems encountered.